File: modul6/C/PRAK605_2310817210017_MUHAMMADNURHUDABAIHAQI.c
#include <stdio.h>
#define N 10 

void multiplyMatrices(int matriksA[N][N], int matriksB[N][N], int hasil[N][N], int n) {
    int i, j, k;
    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
            hasil[i][j] = 0;
            for (k = 0; k < n; k++) {
                hasil[i][j] += matriksA[i][k] * matriksB[k][j];
            }
        }
    }
}
int main() {
    int n, i, j;
    int matriksA[N][N], matriksB[N][N], hasil[N][N];
    printf("\n");
    scanf("%d", &n);

    printf("\n");
    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
            scanf("%d", &matriksA[i][j]);
        }
    }
    printf("\n");
    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
            scanf("%d", &matriksB[i][j]);
        }
    }
    multiplyMatrices(matriksA, matriksB, hasil, n);

    printf("\n");
    printf("Matriks A x B:\n");
    for (i = 0; i < n; i++) {
        for (j = 0; j < n; j++) {
            printf("%d ", hasil[i][j]);
        }
        printf("\n");
    }
    return 0;
}

File: modul6/PYTHON/PRAK605_2310817210017_MUHAMMADNURHUDABAIHAQI.py
def multiply_matrices(matriksA, matriksB, n):
    hasil = [[0 for _ in range(n)] for _ in range(n)]

    for i in range(n):
        for j in range(n):
            for k in range(n):
                hasil[i][j] += matriksA[i][k] * matriksB[k][j]

    return hasil

if __name__ == "__main__":
    n = int(input(""))
    
    print("")
    matriksA = []
    for i in range(n):
        baris = list(map(int, input().split()))
        matriksA.append(baris)
    
    print("")
    matriksB = []
    for i in range(n):
        baris = list(map(int, input().split()))
        matriksB.append(baris)

    hasil = multiply_matrices(matriksA, matriksB, n)
    2
    print("")
    print("Matriks A x B:")
    for baris in hasil:
        print(*baris)
